sklearn/cluster/tests/test_bicluster.py

```diff
@@ -262,3 +262,12 @@ def test_n_features_in_(est):
     assert not hasattr(est, "n_features_in_")
     est.fit(X)
     assert est.n_features_in_ == 3
+
+
+@pytest.mark.parametrize("est", (SpectralBiclustering(), SpectralCoclustering()))
+def test_inf_values_handling_in_(est):
+    X = np.array([[1, 1, 2, 0, 0], [0, 1, 0, 1, 0], [0, 0, 1, 0, 0], [1, 2, 0, 2, 0]])
+
+    with pytest.raises(ValueError, match="array must not contain infs or NaNs"):
+        # this should't fail
+        est.fit(X)
```

Comment:
If there is a 0-sum row, without the mitigation it will not fit and raise an error. Would this be sufficient for the test?

Reply:
In a perfect world, we should be able to check a statistical property on the fitted attributes.
But I have the digged more to be sure.

Reply:
Can you give me a hint what this means for this specific case? I suppose it is related to [your comment](https://github.com/scikit-learn/scikit-learn/pull/29988#discussion_r1803521487) below, but I am still not sure of the direction I could take when thinking how to test better 😬 😁